### DISCUSSION TITLE
Detect if session read-only file status changed

### DISCUSSION THREAD

**Discussion by ArturoDent** [microsoft/vscode-discussions#1225] [Extension Development QnA] [answered]
[cross-posted from Slack]

Is there a way to know if a user has changed the session read-only status of a file?  I can register the read-only commands and that works.

But that doesn't catch when the user attempts to type in a locked file, and clicks on the pop-up link, to set that file to writeable.  That link isn't using any of the three read-only commands.  So I am looking for another way to know either in general when the user changes the session read-only status via command or especially when they do so by clicking the pop-up link (which identifies one of the commands on hover but that command isn't actually used by vscode).

I wish there was an `onDidChangeReadOnlyStatus` event listener...

**Comment by isidorn** [comment, marked as answer]
There is no such listener. I think this is a fair feature request, and you can file one here https://github.com/microsoft/vscode/issues

fyi @bpasero
